### DISCUSSION TITLE
Request body from new route handlers

### DISCUSSION THREAD

**Discussion by djordje-st** [vercel/next.js#46416] [Help] [answered]
Is it possible to get the request body in the new route handlers from 13.2?
I findn't any docs about that and what I did find is that they're not suitable for form submissions but I can't think of a workaround.

~~Tried adding the same route to the `pages` directory but it seems we can't use both methods anymore.~~
Is the current workaround to use API routes in the `pages` directory? If so, how do we make a call from a component to that route instead of the ones in the `app` directory?

**Comment by mike14747** [comment, marked as answer]
With the request body being a ReadableStream, it looks like this is how it's done:

```ts
import { NextRequest, NextResponse } from 'next/server';

export async function POST(request: NextRequest) {
    const body = await request.json();
    console.log({ body });

    // do something with the request body

    return NextResponse.json({ body });
}
```

**Reply by djordje-st** [reply]
That worked, thanks!

**Reply by thomasbln** [reply]
Thanks for the solution - made my day.
But why isn't there anything about it in the "Route Handlers" feature description. ;)
